service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CopyAzureStorageContainerDefinitionStep.java

```diff
@@ -0,0 +1,185 @@
+      throws InterruptedException, RetryException {
+    var inputParameters = flightContext.getInputParameters();
+    var workingMap = flightContext.getWorkingMap();
+    validateInputs(inputParameters, workingMap);
+
+    // get the inputs from the flight context
+    var destinationResourceName =
+        FlightUtils.getInputParameterOrWorkingValue(
+            flightContext,
+            WorkspaceFlightMapKeys.ResourceKeys.RESOURCE_NAME,
+            WorkspaceFlightMapKeys.ResourceKeys.PREVIOUS_RESOURCE_NAME,
+            String.class);
+    var description =
+        FlightUtils.getInputParameterOrWorkingValue(
+            flightContext,
+            WorkspaceFlightMapKeys.ResourceKeys.RESOURCE_DESCRIPTION,
+            WorkspaceFlightMapKeys.ResourceKeys.PREVIOUS_RESOURCE_DESCRIPTION,
+            String.class);
+    var destinationWorkspaceId =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+    var destinationContainerName =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_CONTAINER_NAME, String.class);
+    var destinationResourceId =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_ID, UUID.class);
+    var destStorageAccountId =
+        flightContext
+            .getWorkingMap()
+            .get(
+                WorkspaceFlightMapKeys.ControlledResourceKeys
+                    .DESTINATION_STORAGE_ACCOUNT_RESOURCE_ID,
+                UUID.class);
+
+    ControlledAzureStorageContainerResource destinationContainerResource =
+        buildDestinationControlledAzureContainer(
+            sourceContainer,
+            destStorageAccountId,
+            destinationWorkspaceId,
+            destinationResourceId,
+            destinationResourceName,
+            description,
+            destinationContainerName);
+    ApiAzureStorageContainerCreationParameters destinationCreationParameters =
+        new ApiAzureStorageContainerCreationParameters()
+            .storageContainerName(destinationContainerName)
+            .storageAccountId(destStorageAccountId);
+    ControlledResourceIamRole iamRole =
+        IamRoleUtils.getIamRoleForAccessScope(sourceContainer.getAccessScope());
+
+    // save the container definition for downstream steps
+    workingMap.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.CLONED_RESOURCE_DEFINITION,
+        destinationContainerResource);
+
+    // create the container
+    try {
+      controlledResourceService.createControlledResourceSync(
+          destinationContainerResource, iamRole, userRequest, destinationCreationParameters);
+    } catch (DuplicateResourceException e) {
+      // We are catching DuplicateResourceException here since we check for the container's presence
+      // earlier in the parent flight and bail out if it already exists.
```

Comment:
What is the parent flight of this flight?

Reply:
This is meant to be the "parent flight of this step", will reword.